Add OpenRGB documentation (#41117)
home-assistant/home-assistant.github.io: source/_integrations/openrgb.markdown

---
title: OpenRGB
description: Instructions on how to integrate OpenRGB within Home Assistant.
ha_category:
  - Light
ha_release: "2025.11"
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - "@felipecrs"
ha_domain: openrgb
ha_platforms:
  - light
ha_integration_type: device
ha_quality_scale: silver
---

The **OpenRGB** {% term integration %} is used to integrate the RGB lighting devices from computers running [OpenRGB](https://openrgb.org/).
OpenRGB provides a unified interface for controlling various RGB lighting hardware from different manufacturers.

> 

Reviewer: ```suggestion

## Prerequisites

To use this integration, you need the **OpenRGB application installed** in your computer with the **SDK Server running**.

To enable the OpenRGB SDK Server:

1. In the OpenRGB application, go to **SDK Server** tab.
2. Select **Start Server**.
3. Optionally, go to the **Settings** tab and check **Start at Login**, **Start Minimized**, and **Start Server** for convenience.

```
We usually keep the prereqs before the configuration instructions. See also https://developers.home-assistant.io/docs/documenting/integration-docs-examples#basic-structure-of-an-integration-page, https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:

>   description: The hostname or IP address of the computer running the OpenRGB SDK Server (e.g., `192.168.1.100`).

Reviewer: ```suggestion
  description: The hostname or IP address of the computer running the OpenRGB SDK Server (for example, `192.168.1.100`).
```

Port:
  description: The port on which the OpenRGB SDK Server is accessible (default is `6742`).
{% endconfiguration_basic %}

## Supported devices

This integration supports any RGB device that is compatible with OpenRGB. Refer to the [OpenRGB Supported Devices list](https://openrgb.org/devices.html).

> ## Prerequisites
> 
> To use this integration, you need the **OpenRGB application installed** in your computer with the **SDK Server running**.
> 
> To enable the OpenRGB SDK Server:
> 
> 1. In the OpenRGB application, go to **SDK Server** tab
> 2. Click **Start Server**
> 3. Optionally, go to the **Settings** tab and check **Start at Login**, **Start Minimized**, and **Start Server** for convenience

Reviewer: ```suggestion
```

## Supported devices

This integration supports any RGB device that is compatible with OpenRGB. Refer to the [OpenRGB Supported Devices list](https://openrgb.org/devices.html).

## Supported functionality

### Entities

The OpenRGB integration provides the following entities:

#### Lights

For each RGB device connected to OpenRGB, the integration creates a light entity to control all the LEDs on that device as a single unit.

## Data updates

The **OpenRGB** integration {% term polling polls %} data from the OpenRGB SDK server **every 15 seconds**.

## Reconfiguration

If you need to update your OpenRGB SDK server connection details, you can reconfigure the integration:

1. Go to **{% my integrations title="Settings > Devices & Services" %}**.
2. Select **OpenRGB**. Click the three dots {% icon "mdi:dots-vertical" %} menu and then select **Reconfigure**.
3. Update the hostname/IP address and port number as needed.

> 4. Select **Submit**.

Reviewer 2: This isn't merged right?

Author: No, but here it is:

- https://github.com/home-assistant/core/pull/154478

Reviewer 3: If you remove it from this pull request, then this one can go in.

Author: Right, but there's no big rush (unless https://github.com/home-assistant/core/pull/154478 takes too long to be merged, but I just submitted it).

Also, there's probably no beta coming this week, so no one would actually care, I believe.

Author: PS: I took your comment as a suggestion, not as an ask. I can of course do that if you ask me to.

Reviewer: Please remove this section. https://github.com/home-assistant/core/pull/154478 should come with it's own corresponding documentation PR. https://developers.home-assistant.io/docs/review-process#creating-the-perfect-pr

The integration will reconnect to the OpenRGB SDK Server with the new settings.

## Known limitations

> - Changes made directly in the OpenRGB application may not be immediately reflected in Home Assistant (the integration polls for updates every 15 seconds)
> - Some OpenRGB features like per-LED or per-zone control are not yet supported through Home Assistant

Reviewer: ```suggestion
- Changes made directly in the OpenRGB application may not be immediately reflected in Home Assistant (the integration polls for updates every 15 seconds).
- Some OpenRGB features, like per-LED or per-zone control, are not yet supported through Home Assistant
```

## Troubleshooting

### Connection errors

If you encounter connection errors:

> - In the OpenRGB application, verify **OpenRGB SDK Server is running**
> - In the OpenRGB application, verify that **SDK Server** > **Server Host** is set to `0.0.0.0`
> - Confirm the IP address of the computer running the OpenRGB SDK Server matches the **Host** configured in the OpenRGB integration
> - Confirm the **SDK Server** > **Server Port** matches the port configured in the OpenRGB integration
> - Check for firewall rules that may block the connection from Home Assistant to the OpenRGB SDK Server

Reviewer: ```suggestion
- In the OpenRGB application, verify **OpenRGB SDK Server is running**.
- In the OpenRGB application, verify that **SDK Server** > **Server Host** is set to `0.0.0.0`.
- Confirm the IP address of the computer running the OpenRGB SDK Server matches the **Host** configured in the OpenRGB integration.
- Confirm the **SDK Server** > **Server Port** matches the port configured in the OpenRGB integration.
- Check for firewall rules that may block the connection from Home Assistant to the OpenRGB SDK Server.
```

### Devices not appearing

If your RGB devices don't appear in Home Assistant:

> - Verify the devices are detected and working in the OpenRGB application

Reviewer: ```suggestion
- Verify the devices are detected and working in the OpenRGB application.
```

### Light effects not working

If lighting effects are not working:

> - Verify the effect can be applied in the OpenRGB application

Reviewer: ```suggestion
- Verify the effect can be applied in the OpenRGB application.
```

To enable the OpenRGB SDK server:

1. In the OpenRGB application, go to **SDK Server** tab.
2. Select **Start Server**.
3. Optionally, go to the **Settings** tab and check **Start at Login**, **Start Minimized**, and **Start Server** for convenience.

**Note**: if you opted to **Install System Service** in the OpenRGB **Windows installer**, the SDK server is probably already running. You can check this in the **Services** application in Windows, looking for the **OpenRGB** service.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Name:
  description: A name for this integration entry, like the name of the computer running the OpenRGB SDK server (for example, `My Gaming PC`).
Host:
  description: The hostname or IP address of the computer running the OpenRGB SDK server (for example, `192.168.1.100`).
Port:
  description: The port number that the OpenRGB SDK server is running on (default is `6742`).
{% endconfiguration_basic %}

## Supported devices

This integration supports any RGB device that is compatible with OpenRGB. Refer to the [OpenRGB Supported Devices list](https://openrgb.org/devices.html).

## Supported functionality

> 

Reviewer: ```suggestion
```

---
title: OpenRGB
description: Instructions on how to integrate OpenRGB within Home Assistant.
ha_category:
  - Light
ha_release: 2025.11
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@felipecrs'
ha_domain: openrgb
ha_platforms:
  - light
ha_integration_type: hub
ha_quality_scale: bronze
---

The **OpenRGB** {% term integration %} is used to integrate the RGB lighting devices from computers running [OpenRGB](https://openrgb.org/).
OpenRGB provides a unified interface for controlling various RGB lighting hardware from different manufacturers.

## Prerequisites

> To use this integration, you need the **OpenRGB application installed** in your computer with the **SDK Server running**.

Reviewer: ```suggestion
To use this integration, you need the **OpenRGB** application installed on your computer with the SDK Server running.
```
In general, we try to use bold for UI strings, not for highlighting.

2. Select **Start Server**.
3. Optionally, go to the **Settings** tab and check **Start at Login**, **Start Minimized**, and **Start Server** for convenience.

**Note**: if you opted to **Install System Service** in the OpenRGB **Windows installer**, the SDK server is probably already running. You can check this in the **Services** application in Windows, looking for the **OpenRGB** service.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Name:
  description: A name for this integration entry, like the name of the computer running the OpenRGB SDK server (for example, `My Gaming PC`).
Host:
  description: The hostname or IP address of the computer running the OpenRGB SDK server (for example, `192.168.1.100`).
Port:
  description: The port number that the OpenRGB SDK server is running on (default is `6742`).
{% endconfiguration_basic %}

## Supported devices

This integration supports any RGB device that is compatible with OpenRGB. Refer to the [OpenRGB Supported Devices list](https://openrgb.org/devices.html).

## Supported functionality

### Entities

The OpenRGB integration provides the following entities:

> #### Lights

Reviewer: ```suggestion

The OpenRGB integration provides the following entities:

### Lights
```

Author: Are you sure about this? I think this will break the ToC hierarchy.

Reviewer: > Are you sure about this? I think this will break the ToC hierarchy.

Hm, what do you mean? It's L2, Supported functionality, followed by L3, Lights.
<img width="445" height="261" alt="image" src="https://github.com/user-attachments/assets/d42f4523-5b21-4517-8b4c-804c9f98c097" />

Author: Oops, I didn't realize you removed the _Entities_ L3. Sorry. :D
